### ISSUE TITLE
Abandoning `abook` ... address book alternative(s)?

### ISSUE THREAD

**Issue by ArchieLinux** [neomutt/neomutt#3652]
**QUESTION: Other than `abook` is there another contact-like app (or an internal `NeoMutt` function) I can use to import my 500+ email contacts without having to enter each one manually?** 

After way  too many hours searching for a way  to get (seemingly unsupported) `abook` to accept contact imports properly, I finally surrender. None of the "solutions" I found actually seem to work (for me - user error?!). 

Are current `NeoMutt` users really using `abook`? Or are other approaches available?

I really want to use `NeoMutt` and have configured it almost exactly as I'd like to deploy it ... but I must find some way to incorporate my hundreds of email contacts accumulated over the years. They're available on google contacts, on "cardbook" within Thunderbird - either of which allows me to export in various ways..

For me, not being able to do this would be a showstopper - which I now recall was why I gave up on `NeoMutt` some time ago before recently trying again.

Any suggestions would be greatly appreciated!

**Comment by mschilli87** [comment]
I am using vdirsyncher to sync my contacts via DAV with my Nextcloud instance. It works like a charm. I don't use any Google product and won't help you to figure out how to do anythin with any of their products, but AFAIK vdirsyncher supports importing calendars from their address book service. Simply refer to their documentation. If you struggle to setup NeoMutt to work with vdirsyncher, I'd suggest a search engine of your choice (I guess Google in you case) and if that isn't sufficient, try the NeoMutt mailing list as that would rather be a support request than an issue with NeoMutt that requires fixing (or even attention by the developers). Don't forget to close this issue if you are done. :wink:

**Comment by ArchieLinux** [comment]
Thank you for your kind reply. I will certainly spend some time checking out vdirsyncer. I'm not a big fan of google, either, and don't typically use gmail, but I find import/export contacts through google has made such connections possible over the years.

If I cannot make vdirsyncer work with neomutt, it'll probably not accomplish much else for me in this case, as my only obstacle to using NeoMutt at present is the address book function (abook seems to be atrocious at easily importing common formats).

Either way, when I either solve this or not, I'll close out this request for help. Thanks again!

**Comment by mschilli87** [comment]
It definitely works with NeoMutt as I use such a setup. I just forgot the details about how to configure both tools since I set them up years ago and it just works since. That's why I suggested the mailing list in case you can't figure it out by yourself as somebody there will know and you'll likely reach more people on a user mailing list than on a developer big tracker.

CardDAV is a standard for synchronizing contacts and there any many FOSS-alternatives to Google supportig it.
Once you have imported your contacts via vdirsyncher you could try them out and easily push your contact to them via vdirsyncher in case you want to migrate your address book away from Google.
Also, you might want to checkout khard, which is a neat little tool that plays nicely with vdirsyncher and allows you to query / interact with your (local copy of your) address book from the command line (similar as khal does for calendars).

**Comment by gahr** [comment]
I use a query command that connects to my carddav server. No sync’ing, just the real thing :) 

The quey script is [here](https://github.com/gahr/tcl-dav/blob/master/bin/mutt-query.tcl) and requires the Tcl package from the same repo.

**Comment by mschilli87** [comment]
@gahr: Thx. That's a nice solution as well. Personally I like having my contact available offline on my mobile devices so I guess I'll stick to synching for tjose. But for always-connected workstations this looks like a nice alternative.

**Comment by ArchieLinux** [comment]
I really appreciate the kind replies and suggestions! 

So, I mucked around further and managed to spend more valuable time than advisable. My NeMutt up and running is to use vim-like keys to navigate around the email interface.

I just recently discovered that unlike `thunderbird`, `kmail` offers highly customizable key bindings. In fact, I've been able to create all of the essential `NeoMutt` default keybindings within `kmail` - which addresses one of my primary issues.

It also lets me have excellent control over how HTML, web links, attachments, etc are handled (opened, in-line vs attachment, browser-friendly, etc.)

I assume one can achieve similar customizations within `NeoMutt` but I haven't devoted the time. The whole `abook` obstacle simply got in the way.

At least for now (unless/until I discover some showstopper issues with `kmail` I'll see if it might be the best solution for my use case.

Meanwhile, although a TUI-based email app would be neat and consistent with much of my other daily driver programs, for now I seem to still be stuck in the GUI world when it comes to email.

Thanks again to those above for trying to nudge me along! :+1:
